platforms/defaults_windows.go

```diff
@@ -78,14 +81,21 @@ func revision(v string) int {
 	return r
 }
 
+func prefix(v string) string {
+	parts := strings.Split(v, ".")
+	if len(parts) < 4 {
+		return v
+	}
+	return strings.Join(parts[0:2], ".")
```

Comment:
This will only return the first 2 elements of parts, if the length is >= 4. Which means you'll match ```10.0.20348``` as well as ```10.0.17763```. I _think_ this should be ```return strings.Join(parts[0:3], ".")```

Reply:
Why cant we just match the build number. That is all that is required for process isolation compat right? There is never going to be a 10.0.17763 and a 11.0.17763 :).

Reply:
Sure, that also works. If you believe that the build number will always be unique, whatever works to match the image in the manifest to the version of windows that's running. However, I believe that elsewhere in the code, the major, minor and build numbers are used to compare versions wherever needed.

I pointed out the missing build nr here, because a prefix generated by this would match both Windows Server 2019 and Windows Server 2022. (```10.0```)

Reply:
Sorry @gabriel-samfira for the confusion - You are 100% correct. I agree with you if we do not just use build numbers.

I was asking @jsturtevant if we should just use build numbers since that is the kernel compat level. But its likely just safer to do the major.minor.build